libs/application/templates/new-primary-school/src/fields/DynamicDisabledText/index.tsx

```diff
@@ -0,0 +1,68 @@
+import { FieldComponents, FieldTypes } from '@island.is/application/types'
+import { TextFormField } from '@island.is/application/ui-fields'
+import { useLocale } from '@island.is/localization'
+import { FC, useEffect } from 'react'
+import { useFormContext } from 'react-hook-form'
+import { Application, Field } from '@island.is/application/types'
+import { Locale } from '@island.is/shared/types'
+import { MessageDescriptor } from 'react-intl'
+
+interface DynamicDisabledTextProps {
+  application: Application
+  field: Field & {
+    props: {
+      descriptor:
+        | MessageDescriptor
+        | ((application: Application) => MessageDescriptor)
+      values?: (
+        application: Application,
+        lang: Locale,
+      ) => {
+        [key: string]: string
+      }
```

Comment:
Do I need both of these values? Can't we use the `FormTextWithLocale` type and `formatTextWithLocale` for formatting? 🤔 

```suggestion
      defaultValue: FormTextWithLocale
```

Reply:
You are completely right, have no idea why I did this with two values. Also didn't know about `FormTextWithLocale`. Have changed it